backend/src/federation/CommunityCommunication.ts

```diff
@@ -0,0 +1,43 @@
+import { apiPost } from '@/apis/HttpRequest'
+import CONFIG from '@/config'
+import { Jwt } from 'jsonwebtoken'
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const sodium = require('sodium-native')
+
+/**
+ * create signature from message with community ed25519 private key
+ * @param message as buffer in binary format
+ * @returns buffer containing signature in binary format
+ */
+function signWithCommunityPrivateKey(message: Buffer): Buffer {
+  const sign = Buffer.alloc(sodium.crypto_sign_BYTES)
+  // ed25519 private or secret key starts with 32 Bytes secret and end with 32 Bytes public key
+  const secretKey = Buffer.from(CONFIG.COMMUNITY_PRIVATE_KEY + CONFIG.COMMUNITY_PUBLIC_KEY, 'hex')
+  sodium.crypto_sign_detached(sign, message, secretKey)
+  return sign
+}
+
+/**
+ * start request to open communication with another community or blockchain connector
+ * if succeed, it returns a valid jwt token for further requests
+ * @param communityBPublicKey target community public key in hex format
+ * @param communityBUrl target community url
+ * @returns
+ */
+async function openCommunication(
```

Comment:
No it isn't a test implementation. 
It is only a single part of the communication process. The request send to the Service: openCommunication. Maybe the choosing of my function name wasn't clear enough.  

So the function send a request to another community or blockchain connector for open up a connection. It is assumed that both communication partner are knowing each other. Either by handshaking before or writing each others public key into config. 

How can I verify a JWT token without knowing the secret? As far as I understand JWT, only the issuer of the JWT token need to verify it, if a client make a request with it.
But I can decode it and check the date that it is still valid. I will add this. 

On Blockchain-Connector Side I can save the Community Server Public Key in Config and can so compare it with the received key. When I check the signature I know the sender must have control over the private key or he couldn't have made the signature. I assume that the private key is only known to the server (hence the name private key or secret key). 
What is also possible don't send the own public key with the request so the receiver can only verify the signature if he knows the public key of the sender.

But to be honest the blockchain connector don't really need to know his communication partner at all. 
I only use this call upfront because Moriz has wished for, but it isn't really needed her. (For Blockchain-Connector Communication). I'm not primarily securing the channels, I'm securing the payload. I will explain it more in the next pull request.